Frontend/src/Components/Graph/CustomGraph.js

```diff
@@ -306,50 +287,24 @@ export default function CustomGraph(props) {
         onSave={onSelectSave}
       />
     );
-  }, [
-    isDataSelectOpen,
-    onDataSelectClose,
-    datasetKeys,
-    setDatasetKeys,
-    setHistoryLength,
-    historyLength,
-  ]);
-
-  // the modal that appears when the user saves the graph,
-  // but the graph has no name associated with it
+  }, [isDataSelectOpen, onDataSelectClose, datasetKeys, historyLength, title, onSave]);
+
+
   const graphNameModal = useMemo(
     () => (
       <GraphNameModal
         isOpen={isNameModalOpen}
         onClose={onNameModalClose}
-        onSave={onNewSave}
+        onSave={(name)=>onSave(name, true, { datasets: datasetKeys, historyLength})}
       />
     ),
-    [isNameModalOpen, onNameModalClose, onNewSave]
+    [isNameModalOpen, onNameModalClose]
   );
 
-  // the actual graph shown to the user
-  const graph = useMemo(() => {
-    // const min = DateTime.now().minus(historyLength * 1_000);
-    console.time("graph update");
-    const temp = (
-      <Line
-        data={{ datasets: formattedDatasets }}
-        options={getOptions(
-          DateTime.fromFormat(latestTime, "HH:mm:ss.SSS") ?? DateTime.now(),
-          historyLength,
-          colorMode
-        )}
-        parsing="false"
-      />
-    );
-    console.timeEnd("graph update");
-    return temp;
-  }, [formattedDatasets, historyLength, latestTime]);
 
-  return (
+  return(
     <>
-      <HStack w="100%" h="100%" align="stretch" {...stackProps}>
+      <HStack w="100%" h="100%" align="stretch">
```

Comment:
Why'd we remove this? This might be another issue causing the graph vertical overflow